core/src/mbc/mbc1.rs

```diff
@@ -0,0 +1,137 @@
+    current_bank: u8,
+
+    /// RAM/ROM mode. `false` is ROM mode (bits 5 and 6 in `current_bank` are
+    /// used to select the ROM bank) and `true` is RAM mode (bits 5 and 6 in
+    /// `current_bank` are used to select the RAM bank (0 to 3)).
+    ram_mode: bool,
+
+    /// Whether or not the RAM is enabled.
+    ram_enabled: bool,
+}
+
+
+impl Mbc1 {
+    pub(crate) fn new(data: &[u8], rom_size: RomSize, ram_size: RamSize) -> Self {
+        assert!(rom_size <= RomSize::Banks128, "More than 128 banks, but only MBC1!");
+        assert!(
+            rom_size.len() == data.len(),
+            "Length of cartridge doesn't match length specified in ROM size header",
+        );
+
+        let rom: Vec<_> = data.iter().cloned().map(Byte::new).collect();
+        let ram = vec![Byte::zero(); ram_size.len()];
+
+        Self {
+            rom: rom.into_boxed_slice(),
+            ram: ram.into_boxed_slice(),
+            current_bank: 1,
+            ram_mode: false,
+            ram_enabled: false, // TODO: is that the correct initial value?
+        }
+    }
+
+    /// Returns the real ROM bank number (with respect to `ram_mode`)
+    fn rom_bank(&self) -> usize {
+        if self.ram_mode {
+            (self.current_bank & 0b0001_1111) as usize
+        } else {
+            (self.current_bank & 0b0111_1111) as usize
+        }
+    }
+
+    /// Returns the real RAM bank number (with respect to `ram_mode`)
+    fn ram_bank(&self) -> usize {
+        if self.ram_mode {
+            ((self.current_bank & 0b0110_0000) >> 5) as usize
+        } else {
+            0
+        }
+    }
+}
+
+impl Mbc for Mbc1 {
+    fn load_rom_byte(&self, addr: Word) -> Byte {
+        match addr.get() {
+            // Always bank 0
+            0x0000..0x4000 => self.rom[addr.get() as usize],
+
+            // Bank 1 to N
+            0x4000..0x8000 => {
+                let bank_offset = self.rom_bank() * 0x4000;
+                let relative_addr = addr.get() as usize - 0x4000;
+
+                // We made sure that the actual cartridge data length matches
+                // the number of banks specified in the header. However, the
+                // game might enable a bank higher than specified in the
+                // header. In that case we return FF.
+                self.rom.get(bank_offset + relative_addr)
+                    .cloned()
+                    .unwrap_or(Byte::new(0xFF))
+            }
+
+            _ => unreachable!(),
+        }
+    }
+
+    fn store_rom_byte(&mut self, addr: Word, byte: Byte) {
+        match addr.get() {
+            // RAM enable
+            0x0000..0x2000 => self.ram_enabled = byte.get() & 0x0F == 0x0F,
+
+            // Lower 5 bits of ROM bank number
+            0x2000..0x4000 => {
+                // Again, we can never write 0 to those bits.
+                let new = max(byte.get() & 0b0001_1111, 1);
+                self.current_bank = (self.current_bank & 0b1110_0000) | new;
```

Comment:
Nope. In this part we only write the lower 5 bits (values `0` to `0x1F`). We make sure that we never write 0 into those 5 bits (see the `max()` expression above). This automatically works for `0x20`, `0x40` and `0x60`, too, because those values are only reached by using the additional two bits. 

Mhh.. does this make sense? I feel like this is not a good explanation :/